packages/core/src/package-graph/__tests__/package-graph.spec.ts

```diff
@@ -341,6 +342,103 @@ describe('PackageGraph', () => {
       expect(pkg4.localDependencies.get('pkg-2').fetchSpec).toBe(''); // not found in global catalog so it will show warning below
       expect(logWarnSpy).toHaveBeenCalledWith('graph', 'No version found in "default" catalog for "pkg-2"');
     });
+
+    it('resolves dependencies from catalogs.default when not found in main catalog', () => {
+      readWorkspaceCatalogConfig.mockReturnValueOnce({
+        catalog: {
+          'pkg-1': '1.0.0',
+        },
+        catalogs: {
+          default: {
```

Comment:
I thought that when the `default` catalog is created, we can't have the default `catalog:`. So is your approach really valid in pnpm? I never use the explicit `default`, so I'm curious to know if this is really valid to have both defaults...?

Reply:
no I don't think that we need to validate the structure, the package manager will do that for us way before reaching Lerna. However it felt weird to see this invalid structure in a passing test. Can you just update the test with a valid structure?